### ISSUE TITLE
Discarded primary objects are being included in output

### ISSUE THREAD

**Issue by jenna-tomkinson** [CellProfiler/CellProfiler#4857]
**Describe the bug**

I am currently going through my outputted single cells, and I am noticing that there are rows of single cells that have nuclei coordinates where the nuclei should have been discarded due to size in CellProfiler IdentifyPrimaryObjects. 

I can see when inputting the image in my CellProfiler pipeline that the single-cell comes from, it does not include the nuclei coordinates in the mask that I am seeing in my output. It makes no sense to me how it would be possible to find a cell mask at all without the nucleus.

I can confirm that I do have the `Discard objects outside the diameter range` set as `Yes`.

**To Reproduce**

Currently, the images for this project are not public, so there wouldn't be an exact way to reproduce this time.

**Expected behavior**

The expected behavior is that my output from segmentation and feature extraction should not include nuclei that were excluded due to size. Please let me know if there are any other parameters in my pipeline I should check that could be causing this.

**Screenshots**

Here is my data frame to show that there are all of these rows of single-cells with nuclei all of the same size that are all nuclei that should have been discarded:

![image](https://github.com/CellProfiler/CellProfiler/assets/107513215/eef1f713-251b-4f04-81ba-97766d184ab7)

Here is an example from CellProfiler where there is a discarded nucleus at the same coordinates as the row index 14210 (Well E10):

![image](https://github.com/CellProfiler/CellProfiler/assets/107513215/72a6ff4f-cfb8-4571-8ad4-967870071379)

**Desktop (please complete the following information):**
 - OS: Pop_OS! LTS 22.04
 - CellProfiler Version 4.2.4
 - Installed from source
 - Installed in conda environment via pip

**Additional context**

This looks to hopefully not impact anything biologically since all of these examples look to still be nuclei but there is no way for me to see how it segmented the cells given the parameters I have where these nuclei should have been discarded.

**Comment by bethac07** [comment]
Hey Jenna,

Hmmm, weird. Can you send images and pipeline via email? Even one set would be fine.

**Comment by jenna-tomkinson** [comment]
Sounds good! I will sent that over now!

**Comment by jenna-tomkinson** [comment]
I did a bit of digging, and I can say that I was able to replicate what you got with the 14 objects segmented.

I do believe this has to be some sort of conda environment/build issue since the environment that I ran to produce the data frame with the "discarded" objects is now broken, and I can not start the CellProfiler GUI (gives an error that module bioformats is not installed). 

This means I am unable to replicate the pipeline that had segmented those now discarded nuclei. Super weird and not sure what dependency could have caused it. It is weird to see CellProfiler have variable outputs, but I do not think this is a persistent bug.

After rerunning the same environment file, I was able to make a new conda environment to use CellProfiler with so I will be rerunning this data anyway.

Thank you, Beth, for your assistance with this, and I am glad this was not a huge issue! 🎉 I will close this comment and I can come back to it if it seems like there is some persistent error.
